### ISSUE TITLE
AWSLexVoiceButton freezes app on iOS 11

### ISSUE THREAD

**Issue by JoshClarke** [aws/aws-sdk-ios#756]
### To help us solve your problem better, please answer the following list of questions.

* What service are you using?
AWSLexVoiceButton, AWSCore, AWSLex

* In what version of SDK are you facing the problem?
2.6.5

* Is the issue limited to Simulators / Actual Devices?
Simulators or devices running iOS 11

* Can your problem be resolved if you bump to a higher version of SDK?
I believe I'm on the highest version.

* Is this problem related to specific iOS version?
iOS 11 will freeze and crash, iOS 10 will not

* How are you consuming the SDK? CocoaPods / Carthage / Prebuilt frameworks?
Prebuilt frameworks

* Can you give us steps to reproduce with a minimal, complete, and verifiable example? Please include any specific network conditions that might be required to reproduce the problem.

I can reproduce this in two apps in which I have added the AWSLexVoiceButton.  Anytime I try to initialize an AWSLexVoiceButton, the app freezes and eventually crashes.  This can be from code, XIB, or even a storyboard.  Once the app tries to load the AWSLexVoiceButton, it freezes.  This only happens on iOS 11, if I use 10 everything works great.

While frozen, the CPU usage of the devices is pegged at 100% and memory usage continuously rises until the app crashes.

Thank you,
Josh

**Comment by JoshClarke** [comment]
I was able to work around this issue, and figured out how to reproduce it as well.  Just a note, this is only necessary on iOS 11, the code was working fine before 11 came out.

I was trying to put the voice button in the iOS navigation bar, and doing so would freeze the app unless I explicitly added the button through the main queue.  

Old code:
`- (void)viewDidLoad {
    [super viewDidLoad];
    [_lexVoiceButton setDelegate:self];
    self.navigationItem.rightBarButtonItem = [[UIBarButtonItem alloc]initWithCustomView:_lexVoiceButton];
}`


New code:
`- (void)viewDidLoad {
    [super viewDidLoad];
    [self performSelector:@selector(setupLexButton) withObject:nil afterDelay:1];
}`

`- (void)setupLexButton{
    dispatch_async(dispatch_get_main_queue(), ^{
        [_lexVoiceButton setDelegate:self];
        self.navigationItem.rightBarButtonItem = [[UIBarButtonItem alloc]initWithCustomView:_lexVoiceButton];
    });
}`

**Comment by kredden** [comment]
Do you have your button next to one or more Flexible Space Bar Button Items?

Kevin

**Comment by muellerfr** [comment]
This issue has been closed because of inactivity. Please open a new issue if are still encountering problems.

**Comment by jtarun4625** [comment]
Facing the same problem.

**Comment by JoshClarke** [comment]
Not sure if you'll get this email reply, but here is what I did to fix it:

I was able to work around this issue, and figured out how to reproduce it
as well. Just a note, this is only necessary on iOS 11, the code was
working fine before 11 came out.

I was trying to put the voice button in the iOS navigation bar, and doing
so would freeze the app unless I explicitly added the button through the
main queue.

Old code:
- (void)viewDidLoad { [super viewDidLoad]; [_lexVoiceButton
setDelegate:self]; self.navigationItem.rightBarButtonItem =
[[UIBarButtonItem alloc]initWithCustomView:_lexVoiceButton]; }

New code:
- (void)viewDidLoad { [super viewDidLoad];
dispatch_async(dispatch_get_main_queue(), ^{ [_lexVoiceButton
setDelegate:self]; self.navigationItem.rightBarButtonItem =
[[UIBarButtonItem alloc]initWithCustomView:_lexVoiceButton]; }); }
